core/src/main/java/com/amplifyframework/core/Amplify.java

```diff
@@ -96,7 +101,7 @@ private Amplify() {
         categories.put(CategoryType.HUB, Hub);
         categories.put(CategoryType.DATASTORE, DataStore);
         categories.put(CategoryType.PREDICTIONS, Predictions);
-        return categories;
+        return Immutable.of(categories);
```

Comment:
Ah crap.

https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableMap(java.util.Map)

Not sure how you'll actually return a unmodifiable `LinkedHashMap`, though. You probably don't want to implement your own, that sounds like overkill. 

Hmmm.